stdlib/_typeshed/__init__.pyi

```diff
@@ -179,6 +190,39 @@ class SupportsItemAccess(Protocol[_KT_contra, _VT]):
     def __setitem__(self, key: _KT_contra, value: _VT, /) -> None: ...
     def __delitem__(self, key: _KT_contra, /) -> None: ...
 
+# Protocol for sequence-like objects. This includes commonly used methods
+# from collections.abc.Sequence, and can be used in argument types when using
+# more specific protocols would be cumbersome.
+#
+# Please note that this protocol is not yet marked as stable, and may be
+# extended in the future to include more methods.
+class SequenceLike(Protocol[_T_co]):
+    def __contains__(self, value: object, /) -> bool: ...
+    @overload
+    def __getitem__(self, index: int, /) -> _T_co: ...
+    # This does not necessarily return a sequence of the same type.
+    @overload
+    def __getitem__(self, index: slice, /) -> Sequence[_T_co]: ...
```

Comment:
If it *does* return a sequence of the same type, and that sequence is a `SequenceLike` but not a `Sequence`, then this would be a bit of a problem.
So how about this?


```suggestion
    # This does not necessarily return a sequence of the same type.
    @overload
    def __getitem__(self, index: slice, /) -> SequenceLike[_T_co]: ...
```